File: backend/src/main/java/com/roadsaathi/backend/dto/DriverSessionRequest.java
package com.roadsaathi.backend.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class DriverSessionRequest {

    @NotBlank
    private String fcmToken;

    private Double latitude;
    private Double longitude;
    private Float heading;
    private Float speedKmh;
}

File: android/app/src/main/java/com/roadsaathi/data/remote/dto/DriverSessionRequest.kt
package com.roadsaathi.data.remote.dto

import com.google.gson.annotations.SerializedName

data class DriverSessionRequest(
    @SerializedName("fcm_token")
    val fcmToken: String,
    @SerializedName("latitude")
    val latitude: Double,
    @SerializedName("longitude")
    val longitude: Double,
    @SerializedName("heading")
    val heading: Float? = null,
    @SerializedName("speed_kmh")
    val speedKmh: Float? = null
)
